libtiny_tui/src/input_area/mod.rs

```diff
@@ -659,6 +737,21 @@ impl TextField {
             self.move_cursor(cursor + completion_len as i32);
         }
     }
+
+    fn should_scroll(&self) -> bool {
+        self.scroll.is_some()
+    }
+
+    /// Get the scroll value based on width of TextArea
+    /// If the width is less than the width defined on SCROLL_FALLBACK_WIDTH
+    /// we will transfer to scroll mode, otherwise we will continue to wrap text.
+    fn get_scroll_for_resize(&mut self) -> Option<i32> {
```

Comment:
It would be nice to have a way to fall back to scrolling when the TextField height is greater than, say, half of the entire window height, but I don't think it would be a clean implementation. On resize, we invalidate the height and then calculate it later while drawing.

I think this is a decent way for now, unless someone uses a really wide and short window 😆